### PR TITLE
Elevator encoder settings- added limit switch to elevatorarmsubsystem

### PR DESCRIPTION
**Pull request by Vultimate1**

### CODE REVIEW

File: src/main/java/frc/team4909/robot/operator/generic/BionicJoystick.java [hunk 1]
```diff
@@ -45,38 +46,38 @@ public double getSensitiveAxis(BionicAxis axis) {
...
-        newButton.whenActive(commandable);
+        newButton.whenActive(command);
     }
 
     /**
      * @param button      Button to Create Handler For
      * @param commandable Returns a Commandable that can be used by the operator and autonomous CommandGroups
      */
-    public void buttonHeld(BionicButton button, Commandable commandable) {
+    public void buttonHeld(BionicButton button, InstantCommand commandable) {
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/operator/generic/BionicJoystick.java]: This doesn't seem right. For `buttonPressed`, an `InstantCommand` seems appropriate, but for `buttonHeld`, shouldn't the `Command` run as long as the button is held?

File: src/main/java/frc/team4909/robot/sensors/PhotoElectricSensors.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
+package frc.team4909.robot.sensors;
+import frc.team4909.robot.Robot;
+import edu.wpi.first.wpilibj.DigitalInput;
+public class PhotoElectricSensors{  //not sure if extends or implements
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/sensors/PhotoElectricSensors.java]: `implements` means that there is some `Interface` that dictates what methods you must implement in your class, to conform to that `Interface`.  Below, you are calling `super` in the constructor, which indicates that there is, in fact, a superclass... which means that this class must `extend` some class (which becomes the superclass).

File: src/main/java/frc/team4909/robot/sensors/PhotoElectricSensors.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        frontLeftSensor = new DigitalInput(0);
+        BackLeftSensor = new DigitalInput(1);
+        frontRightSensor = new DigitalInput(2);
+        BackRightSensor = new DigitalInput(3);
+    }
+
+    public void onLine(){
+        
+    if(!frontLeftSensor.get() && !frontRightSensor.get()){
+        state=true;
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/sensors/PhotoElectricSensors.java]: `state` is not a variable name that should have boolean values. What does it mean for `state` to be true? The variable name should be named such that it is obvious, when you set it to a boolean value, what that means.

Comment (derrell) [reply src/main/java/frc/team4909/robot/sensors/PhotoElectricSensors.java]: Besides that, it doesn't appear that either `state` or `state2` are used in this class, so the whole `if` clause doesn't do anything...

File: src/main/java/frc/team4909/robot/sensors/PhotoElectricSensors.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    //  }  
+    //  if(!frontMiddleRightSensor.get()){
+    //      System.out.println("Middle Right Sensor is ON LINE");
+    //      lineFollow();
+    //     }
+     
+    }
+    
+    
+    //Case 1: Coming in from right side at an angle
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/sensors/PhotoElectricSensors.java]: There is only one case here. Obsolete comment?

File: src/main/java/frc/team4909/robot/sensors/PhotoElectricSensors.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    public void lineFollow(){
+      boolean backleft = BackLeftSensor.get();
+      boolean backright = BackRightSensor.get();
+      boolean frontleft = frontLeftSensor.get();
+      boolean frontright = frontRightSensor.get();
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/sensors/PhotoElectricSensors.java]: Do we have four sensors on the robot? Or really, do we have any intention of putting four sensors on the robot?

File: src/main/java/frc/team4909/robot/sensors/PhotoElectricSensors.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+              leftspeed=0.2;
+              rightspeed=0;
+            } if (frontright){
+              System.out.println(leftspeed + "  " + rightspeed + " off line to frontright");
+              rightspeed=0.2;
+              leftspeed=0;
+            }
+            Robot.myDrive.tankDrive(leftspeed, rightspeed);
+        }
+    public static void case1(){
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/sensors/PhotoElectricSensors.java]: Empty function, and never called?

File: src/main/java/frc/team4909/robot/subsystems/ExampleSubsystem.java [hunk 7]
```diff
@@ -9,16 +9,69 @@
 
 import edu.wpi.first.wpilibj.command.Subsystem;
 
+
 /**
  * An example subsystem.  You can replace me with your own Subsystem.
  */
-public class ExampleSubsystem extends Subsystem {
+public class LIDAR extends Subsystem {
```
Comment (derrell) [inline src/main/java/frc/team4909/robot/subsystems/ExampleSubsystem.java]: The class name and the file name do not match.

**Comment by roshanr10** [comment]
hmm thought I closed this, this is a duplicate of #109 with stale code from last month... @Vultimate1 pulled out the relevant change and created #109 instead during todays meeting.

**Pull request closed without merging**